logdevice/common/test/NodesConfigurationPollerTest.cpp

```diff
@@ -0,0 +1,92 @@
+ */
+#include "logdevice/common/NodesConfigurationPoller.h"
+
+#include <algorithm>
+
+#include <folly/Random.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "logdevice/common/ObjectPoller.h"
+#include "logdevice/common/RandomNodeSelector.h"
+#include "logdevice/common/configuration/Configuration.h"
+#include "logdevice/common/configuration/nodes/NodesConfigurationCodec.h"
+#include "logdevice/common/test/TestUtil.h"
+#include "logdevice/common/toString.h"
+
+using namespace facebook::logdevice;
+
+namespace {
+
+class MockNodesConfigurationPoller : public NodesConfigurationPoller {
+ public:
+  virtual ~MockNodesConfigurationPoller() {}
+
+  MockNodesConfigurationPoller(
+      Poller::Options options,
+      VersionExtFn version_fn,
+      Callback cb,
+      folly::Optional<u_int32_t> node_order_seed,
+      folly::Optional<Version> conditional_base_version)
+      : NodesConfigurationPoller(std::move(options),
+                                 std::move(version_fn),
+                                 std::move(cb),
+                                 node_order_seed,
+                                 std::move(conditional_base_version)) {}
+
+ protected:
+  Poller::RequestResult sendRequestToNode(Poller::RoundID round,
+                                          node_index_t node) override {
+    return Poller::RequestResult::OK;
+  }
+
+  std::shared_ptr<const configuration::nodes::NodesConfiguration>
+  getNodesConfiguration() const override {
+    auto node_config = createSimpleNodesConfig(1 /* 1 node */);
+    auto server_config = ServerConfig::fromDataTest(
+        "random_node_selector_test", std::move(node_config));
+
+    auto nodes = server_config->getNodes();
+    auto node_id =
+        NodeID(nodes.begin()->first, nodes.begin()->second.generation);
+    return server_config->getNodesConfigurationFromServerConfigSource();
+  }
+
+  MOCK_METHOD0(getMyNodeID, folly::Optional<node_index_t>());
+  MOCK_METHOD0(getClusterState, ClusterState*());
+  
+
+  // return true if the NCPoller is running in the bootstrapping environment
+  // (e.g., NodesConfigurationInit). If so, conditional polling will be disabled
+  bool isBootstrapping() const override {
+     return false;
+  }
+
+ public:
+  auto testFunction() {
+    return *buildPollingSet({1, 2, 3, 4, 5}, {}, {}, {}, 1, 0).begin();
+  }
+};
+
+TEST(NodesConfigurationPoller, simple) {
+  NodesConfigurationPoller::Poller::Options options;
+  options.mode = NodesConfigurationPoller::Poller::Mode::ONE_TIME;
+  auto mockNodesConfigurationPoller = MockNodesConfigurationPoller(
+      options,
+      configuration::nodes::NodesConfigurationCodec::extractConfigVersion,
+      [this](Status st,
+              NodesConfigurationPoller::Poller::RoundID /*round*/,
+              folly::Optional<std::string> config_str) {},
+      12 /* hard-coded seed value*/,
+      folly::none);
+
+      EXPECT_EQ(3, mockNodesConfigurationPoller.testFunction());
```

Comment:
nit: indentation is not correct.

What about calling the same function multiple times (manually without a loop) to make sure that you'll get the same result?